lib/workers/qemu.js

```diff
@@ -57,9 +53,27 @@ module.exports = class QEMUWorker {
       '-smp', '4'
     ])
 
-    childKillEventHandler(this.child, 'exit')
-    childKillEventHandler(this.child, 'uncaughtException')
-    childKillEventHandler(this.child, 'unhandledRejection')
+    this.child.stderr.on('data', (data) => {
+      stderr += data
+    })
+
+    this.child.on('close', () => {
+      if (stderr.length !== 0) {
+        throw new Error(stderr)
+      }
+    })
+
+    process.once('exit', (code) => {
```

Comment:
You should make sure to remove these handlers in the `off()` function, otherwise they will remain registered even long after the worker completed its job

Reply:
That is a good point.